onboarding/source/amb_light_service.c

```diff
@@ -35,19 +48,86 @@ obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
 
+    if (lightServiceTaskHandle == NULL) {
+        lightServiceTaskHandle = xTaskCreateStatic(
+                                                lightServiceTask,
+                                                LIGHT_SERVICE_NAME,
+                                                LIGHT_SERVICE_STACK_SIZE,
+                                                NULL,
+                                                LIGHT_SERVICE_PRIORITY,
+                                                lightServiceTaskStack,
+                                                &lightServiceTaskBuffer);
+    }
+
+    if (lightServiceTaskHandle == NULL) {
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+    }
+
+    if (lightServiceQueueHandle == NULL) {
+        lightServiceQueueHandle = xQueueCreateStatic(
+                                                    LIGHT_SERVICE_QUEUE_LENGTH,
+                                                    LIGHT_SERVICE_ITEM_SIZE,
+                                                    lightServiceQueueStorage,
+                                                    &lightServiceQueue);
+    }
+
+    if (lightServiceQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
+
+    return OBC_ERR_CODE_SUCCESS;
+
     /* USER CODE END */
 }
 
+uint32_t getLightSensorData(void) {
+    // TAKEN FROM https://git.ti.com/cgit/hercules_examples/hercules_examples/tree/Launchpad/RM/RM46L8/Project_1/demoapp/source/adc_demos.c
+
+    adcData_t lightData;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &lightData);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (lightData.value);
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
 
+    light_event_t lightEvent; // Buffer into which the event will be copied
+    
+    if (lightServiceQueueHandle != NULL) {
+        if (xQueueReceive(lightServiceQueueHandle, &lightEvent, (TickType_t) 10) == pdPASS) {
+            if (lightEvent == MEASURE_LIGHT) {
+                    uint32_t lightSensorData = getLightSensorData();
+
+                    sciPrintf("Ambient light value measured: %u", lightSensorData);
```

Comment:
For future reference, the format specifier for `uint32_t` values is `%lu`, not `%u`.